pxtlib/emitter/emitter.ts

```diff
@@ -423,8 +426,17 @@ namespace ts.pxtc {
         res.jsDoc = ""
         cmt = cmt.replace(/\/\*\*([^]*?)\*\//g, (full: string, doccmt: string) => {
             doccmt = doccmt.replace(/\n\s*(\*\s*)?/g, "\n")
-            doccmt = doccmt.replace(/^\s*@param\s+(\w+)\s+(.*)$/mg, (full: string, name: string, desc: string) => {
+            doccmt = doccmt.replace(/^\s*@param\s+(\w+)\s+(\[.*\])?\s+(.*)$/mg, (full: string, name: string, type: string, desc: string) => {
                 res.paramHelp[name] = desc
+                if (type) {
+                    type.replace(/^\[([0-9]+)-([0-9]+)\]$/i, (full: string, min: string, max: string) => {
+                        if (!res.min) res.min = {}
+                        if (!res.max) res.max = {}
+                        res.min[name] = min
+                        res.max[name] = max
+                        return ""
```

Comment:
I created a regex to identify the range values in the parameter declaration and store those values